cluster/v1alpha1/types_rolloutstrategy.go

```diff
@@ -0,0 +1,96 @@
+
+// Rollout strategy to be used by workload applier controller.
+type RolloutStrategy struct {
+	// Rollout strategy Types are All, Progressive and ProgressivePerGroup
+	// 1) All means apply the workload to all clusters in the decision groups at once.
+	// 2) Progressive means apply the workload to the selected clusters progressively per cluster. The workload will not be applied to the next cluster unless one of the current applied clusters reach the successful state or timeout.
+	// 3) ProgressivePerGroup means apply the workload to decisionGroup clusters progressively per group. The workload will not be applied to the next decisionGroup unless all clusters in the current group reach the successful state or timeout.
+	// +kubebuilder:default:=All
+	// +optional
+	Type RolloutStrategyType `json:"type,omitempty"`
+
+	// All RolloutStrategyType
+	// +optional
+	All *RolloutAll `json:"all,omitempty"`
+
+	// Progressive RolloutStrategyType
+	// +optional
+	Progressive *RolloutProgressive `json:"progressive,omitempty"`
+
+	// ProgressivePerGroup RolloutStrategyType
+	// +optional
+	ProgressivePerGroup *RolloutProgressivePerGroup `json:"progressivePerGroup,omitempty"`
+}
+
+// RolloutStrategy Types
+type RolloutStrategyType string
+
+const (
+	//All means apply the workload to all clusters in the decision groups at once.
+	All RolloutStrategyType = "All"
+	//Progressive means apply the workload to the selected clusters progressively per cluster.
+	Progressive RolloutStrategyType = "Progressive"
+	//ProgressivePerGroup means apply the workload to the selected clusters progressively per group.
+	ProgressivePerGroup RolloutStrategyType = "ProgressivePerGroup"
+)
+
+// RolloutAll is RolloutStrategyType
+type RolloutAll struct {
+	// +optional
+	Timeout Timeout `json:",inline"`
+}
+
+// RolloutProgressivePerGroup is RolloutStrategyType
+type RolloutProgressivePerGroup struct {
+	// List of the decision groups names or indexes to apply the workload first and fail if workload did not reach successful ((not proceed to apply workload to other decision groups/clusters).
+	// +optional
+	MandatoryDecisionGroups []MandatoryDecisionGroup `json:"mandatoryDecisionGroups,omitempty"`
+
+	// +optional
+	Timeout Timeout `json:",inline"`
+}
+
+// RolloutProgressive is RolloutStrategyType
+type RolloutProgressive struct {
+	// List of the decision groups names or indexes to apply the workload first and fail if workload did not reach successful ((not proceed to apply workload to other decision groups/clusters).
+	// +optional
+	MandatoryDecisionGroups []MandatoryDecisionGroup `json:"mandatoryDecisionGroups,omitempty"`
+
+	// MaxConcurrency is the max number of clusters to deploy workload concurrently. The default value for MaxConcurrency is determined from the clustersPerDecisionGroup defined in the placement->DecisionStrategy.
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +optional
+	MaxConcurrency string `json:"maxConcurrency,omitempty"`
```

Comment:
Have the below question for `maxconcurrency`, might need a clearer explanation.

1. Does the `maxconcurrency` also work on MandatoryDecisionGroup？
2. If a placement strategy is defined as below:
```
  DecisionStrategy:
    numberOfClustersPerDecisionGroup: 150
    DecisionGroups:
    - label: prod-canary-west
      clusterSelector:
        matchExpressions:
          - key: prod-canary-west
            operator: Exist
….
status:
  numberOfSelectedClusters: 300
  decisionGroups:
  - decisionGroupIndex: 0
    decisionGroupLabel: prod-canary-west
    placementDecisions:
    - ztp-placement-decision-0 // cluster1 to cluster 100
    - ztp-placement-decision-1 // cluster101 to cluster120
    clusterCount: 120
  - decisionGroupIndex: 1
    placementDecisions:
    - ztp-placement-decision-2 // cluster121 to cluster 220
    - ztp-placement-decision-3 // cluster221 to cluster 270
    clusterCount: 150 //
  - decisionGroupIndex: 2
    placementDecisions:
    - ztp-placement-decision-4  // cluster271 to cluster 300
    clusterCount: 30
```
When type is `rolloutProgressive` and not define any `mandatoryDecisionGroups`, what’s the expected upgrade behavior, will decision groups be ignored and upgrade cluster1 to cluster 150 first actually?